sim/dex-conditions.ts

```diff
@@ -651,7 +652,7 @@ export class DexConditions {
 	}
 
 	getByID(id: ID): Condition {
-		if (!id) return EMPTY_CONDITION;
+		if (id === '') return EMPTY_CONDITION;
```

Comment:
(any particular reason why you used `id === ''` instead of `!id` as your test?)

Reply:
One part preference for being explicit.

Other part: My mental model is that V8 does better when there are fewer types to contend with.
Truthy/falsy tests require it to handle `boolean false`, `number 0`, `number NaN`, `string empty`, `null`, and `undefined`.  I might have even forgotten one!
We know that only `string` will actually be passed.
Here's a not-terribly-easy-to-read godbolt example https://godbolt.org/z/n45YbPvEn.
What I observed is that the truthy version (middle) does several more type checks (compare and branch) than the left (explicit) version. Ends being a decent difference in function size.

To be clear, the above doesn't matter in 99.999% of JS code.


Reading JIT generated ASM is really something else... (yuck)